acceptancetests/assess_cross_model_relations.py

```diff
@@ -104,6 +111,66 @@ def ensure_cmr_offer_management(client):
         assert_offer_can_be_deleted(management_model, offer_url)
 
 
+def ensure_cmr_offer_consumption_and_relation(first_client, second_client):
+    """Ensure offers can be consumed by another model.
+
+    :param first_client: ModelClient model that will be the source of the
+      offer.
+    :param second_client: ModelClient model that will consume the offered
+      application endpoint.
+    """
+    with temporary_model(first_client, 'relation-source') as source_client:
+        with temporary_model(second_client, 'relation-sink') as sink_client:
+            offer_url, offer_name = deploy_and_offer_db_app(source_client)
+            assert_relating_to_offer_succeeds(sink_client, offer_url)
+            assert_saas_url_is_correct(sink_client, offer_name, offer_url)
+
+
+def ensure_user_can_consume_offer(first_client, second_client):
+    """Ensure admin is able to grant a user access to an offer.
+
+    Almost the same as `ensure_cmr_offer_consumption_and_relation` except in
+    this a user is created on all controllers (might be a single controller
+    or 2) with the permissions of 'login' for the source controller and 'write'
+    for the model the user will deploy an application to consume the offer.
+
+    :param first_client: ModelClient model that will be the source of the
+      offer.
+    :param second_client: ModelClient model that will consume the offered
+      application endpoint.
+    """
+    with temporary_model(first_client, 'relation-source') as source_client:
+        with temporary_model(second_client, 'relation-sink') as sink_client:
+            username = 'theundertaker'
+            source_controller_name = source_client.env.controller.name
+            sink_controller_name = sink_client.env.controller.name
+            with temp_dir() as tmp_dir:
+                # The models might be on the same controller, create a user
+                # only once per controller.
+                user_sink_client = create_user_on_controller(
+                    sink_client, username, tmp_dir, 'sink_controller', 'write')
+                if sink_controller_name != source_controller_name:
+                    create_user_on_controller(
```

Comment:
Am I right in seeing that create_user_on_controller() calls register_user()?
We don't want to register_user() on both controllers.
On the offering controller, we juju add-user.
On the consuming controller, we juju register-user
How does the code as written work? register user requires a user to first be added and then you get a token string to use when registering.